specs/account_spec.rb

```diff
@@ -0,0 +1,163 @@
+      # expression matching one or more characters - as long as
+      # anything at all is printed out the test will pass.
+      proc {
+        account.withdraw(withdrawal_amount)
+      }.must_output /.+/
+    end
+
+    it "Doesn't modify the balance if the account would go negative" do
+      start_balance = 100.0
+      withdrawal_amount = 200.0
+      account = Bank::Account.new(1337, start_balance)
+
+      updated_balance = account.withdraw(withdrawal_amount)
+
+      # Both the value returned and the balance in the account
+      # must be un-modified.
+      updated_balance.must_equal start_balance
+      account.balance.must_equal start_balance
+    end
+
+    it "Allows the balance to go to 0" do
+      account = Bank::Account.new(1337, 100.0)
+      updated_balance = account.withdraw(account.balance)
+      updated_balance.must_equal 0
+      account.balance.must_equal 0
+    end
+
+    it "Requires a positive withdrawal amount" do
+      start_balance = 100.0
+      withdrawal_amount = -25.0
+      account = Bank::Account.new(1337, start_balance)
+
+      proc {
+        account.withdraw(withdrawal_amount)
+      }.must_raise ArgumentError
+    end
+  end
+
+  describe "Account#deposit" do
+    it "Increases the balance" do
+      start_balance = 100.0
+      deposit_amount = 25.0
+      account = Bank::Account.new(1337, start_balance)
+
+      account.deposit(deposit_amount)
+
+      expected_balance = start_balance + deposit_amount
+      account.balance.must_equal expected_balance
+    end
+
+    it "Returns the modified balance" do
+      start_balance = 100.0
+      deposit_amount = 25.0
+      account = Bank::Account.new(1337, start_balance)
+
+      updated_balance = account.deposit(deposit_amount)
+
+      expected_balance = start_balance + deposit_amount
+      updated_balance.must_equal expected_balance
+    end
+
+    it "Requires a positive deposit amount" do
+      start_balance = 100.0
+      deposit_amount = -25.0
+      account = Bank::Account.new(1337, start_balance)
+
+      proc {
+        account.deposit(deposit_amount)
+      }.must_raise ArgumentError
+    end
+  end
+end
+
+# TODO: change 'xdescribe' to 'describe' to run these tests
+xdescribe "Wave 2" do
+  describe "Account.all" do
+    it "Returns an array of all accounts" do
+      # TODO: Your test code here!
+      # Useful checks might include:
+      #   - Account.all returns an array
+      #   - Everything in the array is an Account
+      #   - The number of accounts is correct
+      #   - The ID and balance of the first and last
+      #       accounts match what's in the CSV file
+      # Feel free to split this into multiple tests if needed
+    end
+  end
+
+  describe "Account.find" do
+    it "Returns an account that exists" do
+      # TODO: Your test code here!
+    end
+
+    it "Raises an error for an account that doesn't exist" do
+      # TODO: Your test code here!
+    end
```

Comment:
Maybe something looking for the 1st and last accounts in the file, or maybe we should let them figure that part out.